scripts/discourse-sync/client.py

```diff
@@ -0,0 +1,23 @@
+import pydiscourse
+
+class Client(pydiscourse.DiscourseClient):
+    """Extends pydiscourse.DiscourseClient to support additional API methods."""
+
+    def lock_post(self, post_id: int, locked: bool, **kwargs):
+        """Lock a post from being edited
+
+        https://docs.discourse.org/#tag/Posts/operation/lockPost
+
+        Args:
+            post_id: the ID of the post to lock
+            locked: True to lock, False to unlock
+        """
+        if locked:
+            kwargs["locked"] = "true"
+        else:
+            kwargs["locked"] = "false"
+        return self._put(f"posts/{post_id}/locked.json", **kwargs)
```

Comment:
I guess it's minor:

Given that it's a new method, called only once, without extra keyword arguments, perhaps free-form keyword arguments are not needed?

```suggestion
    def lock_post(self, post_id: int, locked: bool):
        """Lock a post from being edited

        https://docs.discourse.org/#tag/Posts/operation/lockPost

        Args:
            post_id: the ID of the post to lock
            locked: True to lock, False to unlock
        """
        return self._put(f"posts/{post_id}/locked.json", locked="true" if locked else "false")
```

(I guess `locked=str(locked).lower()` would also cut it, though the above is more explicit)

Looking through the upstream source, https://github.com/pydiscourse/pydiscourse/blob/master/src/pydiscourse/client.py additional keyword arguments are allowed, but the content is either not documented or labelled as "who knows".

Reply:
Unrelated to the above, I wonder if `self._put(..., json=True)` is needed.